`AdvancedTable`: Website documentation (#2628)
hashicorp/design-system: website/docs/components/table/advanced-table/partials/accessibility/accessibility.md

Reviewer: [Question] Should we add in some of the accessibility notes from the [basic table docs](https://hds-website-git-hds-4231-advanced-table-docs-hashicorp.vercel.app/components/table/table?tab=accessibility#focus-in-tables) around things like focus and best practices? Or maybe even just add a link to them?

Author: Fair! The focusable cell part is no longer relevant... nor is the display thing because we're manually adding roles to all the parts of the table. But I'll add information about the rows not being focusable.

Author: Updated the section.

## Conformance rating

<Doc::Badge @type="success">Conformant</Doc::Badge>

When used as recommended, there should not be any WCAG conformance issues with this component.

## Best practices

### Interactive rows

The table row element cannot receive interactions, meaning actions cannot be attached directly to a table row. If you need an interactive element, place it within a table cell element in that row (i.e., `<div role="gridcell"><a href="somelink.html">Some link</a></div>`).

### Focus in Advanced Tables

Unlike the Table component, each cell receives focus in the Advanced Table to let users navigate through the table efficiently with a keyboard. For any other interactions, you must use interactive elements (buttons, links, etc.) within the cells. 

### Row selection

> You should clearly communicate to the user how many rows are selected and how many rows there are total outside of the Advanced Table. For additional considerations, read the [Multi-select usability and accessibility considerations](/components/table/advanced-table?tab=code#usability-and-accessibility-considerations).

Author: It means like display a piece of text outside of the advanced table so no matter what page you are currently on you can know how many rows are selected

![Screenshot 2025-01-22 at 5 03 25 PM](https://github.com/user-attachments/assets/75bc6434-9879-487a-aa7e-14138808d251)

> You should clearly communicate to the user how many rows are selected and how many rows there are total outside of the Advanced Table. For additional considerations, read the [Multi-select usability and accessibility considerations](/components/table/advanced-table?tab=code#usability-and-accessibility-considerations).

Author: (this image is from the table multi select pattern: https://helios.hashicorp.design/patterns/table-multi-select)

## Conformance rating

> <!-- Update conformance rating badge with correct status -->

Reviewer 2: ```suggestion
```

<Doc::Badge @type="success">Conformant</Doc::Badge>

When used as recommended, there should not be any WCAG conformance issues with this component.

## Best practices

### Interactive rows

> The table row element is not eligible to receive interactions. That is, actions cannot be attached to a table row. If an interactive element is desired, place it within a table cell element within that row (i.e., `<div role="gridcell"><a href="somelink.html">Some link</a></div>`).

Reviewer 2: ```suggestion
The table row element cannot receive interactions, meaning actions cannot be attached directly to a table row. If you need an interactive element, place it within a table cell element in that row (i.e., `<div role="gridcell"><a href="somelink.html">Some link</a></div>`).
```

### Focus in Advanced Tables

> Unlike the Table component, each cell receives focus in the Advanced Table to let users navigate through the table efficiently with a keyboard. For any other interactions, there must be interactive elements (buttons, links, etc.) within the cells. 

Reviewer 2: ```suggestion
Unlike the Table component, each cell receives focus in the Advanced Table to let users navigate through the table efficiently with a keyboard. For any other interactions, you must use interactive elements (buttons, links, etc.) within the cells. 
```

### Row selection

> It is best to clearly communicate to the user how many rows are selected and how many rows there are total outside of the Advanced Table. For more considerations, read the [Multi-select usability and accessibility considerations](/components/table/advanced-table?tab=code#usability-and-accessibility-considerations).

Reviewer 2: ```suggestion
You should clearly communicate to the user how many rows are selected and how many rows there are total outside of the Advanced Table. For additional considerations, read the [Multi-select usability and accessibility considerations](/components/table/advanced-table?tab=code#usability-and-accessibility-considerations).
```

## Conformance rating

<!-- Update conformance rating badge with correct status -->
<Doc::Badge @type="success">Conformant</Doc::Badge>

When used as recommended, there should not be any WCAG conformance issues with this component.

## Best practices

### Interactive rows

The table row element is not eligible to receive interactions. That is, actions cannot be attached to a table row. If an interactive element is desired, place it within a table cell element within that row (i.e., `<div role="gridcell"><a href="somelink.html">Some link</a></div>`).

### Focus in Advanced Tables

Unlike the Table component, each cell receives focus in the Advanced Table to let users navigate through the table efficiently with a keyboard. For any other interactions, there must be interactive elements (buttons, links, etc.) within the cells. 

### Row selection

It is best to clearly communicate to the user how many rows are selected and how many rows there are total outside of the Advanced Table. For more considerations, read the [Multi-select usability and accessibility considerations](/components/table/advanced-table?tab=code#usability-and-accessibility-considerations).

> if they have selected rows outside of their current view and how many out of the total data set are selected

Reviewer: [Question] Was this meant to be part of the previous sentence or a new note? 👀

Author: Meant to be deleted - oops!
